### PR TITLE
[RPA-552] Select element in browser widget

### PR DESCRIPTION
**Pull request by FranekMarciniak**

### CODE REVIEW

**Comment by FranekMarciniak** [comment]
TODO polskie translacje

File: runbotics/runbotics-orchestrator-ui/src/translations/en/process.json [hunk 1]
```diff
@@ -148,6 +148,20 @@
...
+    "Process.Details.Modeler.Actions.Browser.Target.ID": "ID",
+    "Process.Details.Modeler.Actions.Browser.Target.ClassName": "Class",
+    "Process.Details.Modeler.Actions.Browser.Target.LinkText": "Link text",
+    "Process.Details.Modeler.Actions.Browser.Target.PartialLinkText": "Partial Link Text",
+    "Process.Details.Modeler.Actions.Browser.Target.None": "Custom",
+    "Process.Details.Modeler.Actions.Browser.Target.ID.Tooltip": "Selects an element by its id attribute (e.g. <strong>my-id</strong>",
+    "Process.Details.Modeler.Actions.Browser.Target.XPath.Tooltip": "Selects an element by its XPath (e.g. <strong>//div[@id='my-id']</strong>)",
+    "Process.Details.Modeler.Actions.Browser.Target.CSS.Tooltip": "Selects an element by its CSS selector (e.g. <strong>input[name='username']</strong>, <strong>css=#my-id</strong>)",
+    "Process.Details.Modeler.Actions.Browser.Target.ClassName.Tooltip": "Selects an element by its class name (e.g. <strong>my-class</strong>",
+    "Process.Details.Modeler.Actions.Browser.Target.LinkText.Tooltip": "Selects an link tag by its exact text (e.g. <strong>Click here</strong>)",
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/translations/en/process.json]: maybe change it to `Selects a link html element by its exact text`
and rewrite the below one also

File: runbotics/runbotics-orchestrator-ui/src/translations/en/process.json [hunk 2]
```diff
@@ -148,6 +148,20 @@
...
+    "Process.Details.Modeler.Actions.Browser.Target.PartialLinkText.Tooltip": "Selects an link tag by its partial text (e.g. <strong>Click</strong>)",
+    "Process.Details.Modeler.Actions.Browser.Target.None.Tooltip": "This gives you the ability set prefix yourself (e.g. <strong>css=#my-id</strong>, <strong>xpath=//div[@id='my-id']</strong>)",
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/translations/en/process.json]: change it to `Allows to define selector type yourself`

File: runbotics/runbotics-orchestrator-ui/src/translations/pl/process.json [hunk 3]
```diff
@@ -150,6 +150,20 @@
...
+    "Process.Details.Modeler.Actions.Browser.Target.XPath": "XPath",
+    "Process.Details.Modeler.Actions.Browser.Target.CSS": "CSS",
+    "Process.Details.Modeler.Actions.Browser.Target.ID": "ID",
+    "Process.Details.Modeler.Actions.Browser.Target.ClassName": "Class",
+    "Process.Details.Modeler.Actions.Browser.Target.LinkText": "Link text",
+    "Process.Details.Modeler.Actions.Browser.Target.PartialLinkText": "Partial Link Text",
+    "Process.Details.Modeler.Actions.Browser.Target.None": "Custom",
+    "Process.Details.Modeler.Actions.Browser.Target.ID.Tooltip": "Selects an element by its id attribute (e.g. <strong>my-id</strong>",
+    "Process.Details.Modeler.Actions.Browser.Target.XPath.Tooltip": "Selects an element by its XPath (e.g. <strong>//div[@id='my-id']</strong>)",
+    "Process.Details.Modeler.Actions.Browser.Target.CSS.Tooltip": "Selects an element by its CSS selector (e.g. <strong>input[name='username']</strong>, <strong>css=#my-id</strong>)",
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/translations/pl/process.json]: these texts should be in polish

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+const PREFIX_WIDTH = 170;
+
+interface BrowserTargetState {
+    prefix: string;
+    value: string;
+}
+
+const BrowserTargetWidget: FC<WidgetProps> = (props) => {
+    const [formState, setFormState] = React.useState<BrowserTargetState>({prefix: '', value: ''});
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx]: move initial state to variable

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+    useEffect(() => {
+        if(props.value){
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx]: spaces needed

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+import ElementAwareAutocompleteWidget from './ElementAwareAutocompleteWidget';
+
+const FieldsWrapper = styled.div`
+    display: flex;
+    width: 100%;
+    align-items: center;
+    gap: 16px;
+`;
+
+const PREFIX_OPTIONS = {
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx]: move it to the component or create custom hook to use `useTranslations` hook

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+import { InputLabel, MenuItem, Select, SelectChangeEvent } from '@mui/material';
+import { WidgetProps } from '@rjsf/core';
+import styled from 'styled-components';
+
+import { translate } from 'src/hooks/useTranslations';
+
+import ElementAwareAutocompleteWidget from './ElementAwareAutocompleteWidget';
+
+const FieldsWrapper = styled.div`
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx]: move everything but component to appropriate files like `.types`, `.utils`, `.styles`

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+        onChangeWrapper(event.target.value, formState.value);
+    };
+
+    const handleValueChange = (event: string) => {
+        setFormState(prev => ({ prefix: prev.prefix, value: event ?? '' }));
+        onChangeWrapper(formState.prefix, event ?? '');
+    };
+
+    return <FieldsWrapper>
+        <InputLabel>Selector</InputLabel>
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx]: translation

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+        setFormState(prev => ({prefix: event.target.value, value: prev.value}));
+        onChangeWrapper(event.target.value, formState.value);
+    };
+
+    const handleValueChange = (event: string) => {
+        setFormState(prev => ({ prefix: prev.prefix, value: event ?? '' }));
+        onChangeWrapper(formState.prefix, event ?? '');
+    };
+
+    return <FieldsWrapper>
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx]: use parentheses

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+        <InputLabel>Selector</InputLabel>
+        <Select sx={{ minWidth: `${PREFIX_WIDTH}px`}} required={props.required} label="Selector"  onChange={handlePrefixChange} value={formState.prefix}>
+            {Object.keys(PREFIX_OPTIONS).map((option) => <MenuItem key={option} value={option}>{PREFIX_OPTIONS[option].label}</MenuItem>)}
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx]: move it to constant and brake it to multiple lines

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+
+    useEffect(() => {
+        if(props.value){
+            const prefix = Object.keys(PREFIX_OPTIONS).find((option) => props.value.startsWith(option)) || 'None';
+            const value = prefix === 'None' ? props.value : props.value.replace(prefix, ''); 
+            setFormState({ prefix: prefix, value: value});
+        }
+    }, []);
+
+    const onChangeWrapper = (prefix: string, value: string) => {
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx]: rename to `handleChangeWrapper` sounds like a prop

File: runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+    };
+
+    return <FieldsWrapper>
+        <InputLabel>Selector</InputLabel>
+        <Select sx={{ minWidth: `${PREFIX_WIDTH}px`}} required={props.required} label="Selector"  onChange={handlePrefixChange} value={formState.prefix}>
+            {Object.keys(PREFIX_OPTIONS).map((option) => <MenuItem key={option} value={option}>{PREFIX_OPTIONS[option].label}</MenuItem>)}
+        </Select>
+        <ElementAwareAutocompleteWidget
+            {...props}
+            sx={{flexGrow: 1}}
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/BrowserTargetWidget.tsx]: spaces between curlies and content

**Review by preludium — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-orchestrator-ui/src/translations/en/process.json (hunk 2 above)
Comment (preludium) [reply runbotics/runbotics-orchestrator-ui/src/translations/en/process.json]: maybe it's worth to list all possible selectors?

**Review by preludium — APPROVED** [review]

**Pull request merged**